src/common.nugetversions.targets

```diff
@@ -0,0 +1,97 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!--===============================================================================================================
+    SPECIAL SAUCE
+    ===================================================================================================================  
+    This section contains any special sauce needed to implement Global NuGet Management.-->
+
+  <!--Since Global NuGet Management requires having a single [packages.config] file that's located in a special
+        global location, its existence won't be known by default to Projects when they are opened/built.  As such,
+        we need the following target that runs before Project References are resolved, which is responsible for
+        downloading and using [NuGet.exe] to perform a manual restore operation on the global [packages.config]
+        file.  We run this target only for local developer inner loop scenarios (BuildingInsideVisualStudio=true).  For
+        Azure DevOps builds, a NuGet Restore pipeline task is run against the [packages.config] file instead.-->
+
+  <PropertyGroup>
+    <Terminal_NuGetEXE>$(SolutionDir)dep\nuget\nuget.exe</Terminal_NuGetEXE>
+    <Terminal_GlobalPackagesConfigPath>$(SolutionDir)dep\nuget\packages.config</Terminal_GlobalPackagesConfigPath>
+  </PropertyGroup>
+
+  <Target Name="_Terminal_NuGetPackageManager"
+      BeforeTargets="ResolveProjectReferences"
+      Inputs="$(Terminal_GlobalPackagesConfigPath)"
+      Outputs="$(SolutionDir)packages\packages.config"
+      Condition="'$(BuildingInsideVisualStudio)' == 'true'">

```

Comment:
Ah, interesting. Since this restore (1) only happens inside VS and (2) cannot be automatically detected via a commandline `nuget restore`, are we going to have to update our README/onboarding guide? Or some of the "recommended" build steps in `OpenConsole.psm1`/`razzle.cmd`?

Reply:
I'm not sure, I didn't rely on either of those scripts.  My testing was `git clean -fdx` to wipe the repo to clean, open OpenConsole.sln, change to Debug|x64, and then F7 to build.  

What sort of nuget detection are you referring to?  Wouldn't the existence of the packages in the correct location satisfy it?  Or is there some other subtlety I'm not aware of?

Reply:
Ah, I was thinking of `nuget restore openconsole.sln`... since nuget would originally look for `packages.config` projects and explicitly restore them, i was worried that that would no longer work since we dropped `packages.config` from the solution's project/package closure.

Reply:
As far as I know that will restore packages.config for any projects in the solution.  That still has a few things (just the two XAML 2.7 versions).  Those don't overlap with the global packages.config so I think it is fine.

Reply:
But then.. if you're building from nothing outside of VS (this new restore step requires `BuildingInsideVisualStudio` to be true), we need to tell folx to additionally restore `build/packages.config` or somesuch right?

Reply:
Ahhh yes you are correct.  A new restore call will need to be added (like the YML files) or a different variable that I can condition off.  Is there any sort of "ADO pipeline" variable I could use instead of BuildingInsideVisualStudio?

Reply:
I typically look for `$(AGENT_ID)`. If it's empty, we're not on ADO.

After this fix, I suspect that testing with `msbuild /m openconsole.sln blah blah blah` will be enlightening :smile:

Reply:
After thinking about it I think fixing up the razzle/pwsh build environments would be simpler.  I just had to add a second restore call to pull everything in and they should work fine.  (I'm running a build now to confirm but the restore worked as expected with the tweak and was broken by the previous changes so I think it will be fine).

Reply:
Both builds (razzle.cmd and Invoke-OpenConsoleBuild) succeeded after a `git clean -fdx` to wipe to a clean repo.